keel-core/src/main/kotlin/com/netflix/spinnaker/keel/core/api/PromotionStatus.kt

```diff
@@ -1,5 +1,38 @@
 package com.netflix.spinnaker.keel.core.api
 
 enum class PromotionStatus {
-  PENDING, APPROVED, DEPLOYING, CURRENT, PREVIOUS, VETOED
+  /**
+   * Waiting on constraint evaluation before being approved
+   */
+  PENDING,
+
+  /**
+   * Has passed constraints successfully, and wil be deployed imminently
+   */
+  APPROVED,
+
+  /**
+   * Deploying in the environment
+   */
+  DEPLOYING,
+
+  /**
+   * Currently deployed in the environment
+   */
+  CURRENT,
+
+  /**
+   * Was deployed in the environment, but was replaced by a new version
+   */
+  PREVIOUS,
+
+  /**
+   * Never allowed to be in the environment again
+   */
+  VETOED,
+
+  /**
+   * Was approved for the environment, but a newer version was deployed instead
+   */
+  SUPERSEDED
```

Comment:
fwiw I will most likely continue to use "skipped" in the UI — while in isolation superseded is the most precise term for this concept, when taken in the context of it living alongside the "previous" status it gets pretty ambiguous. Part of the meaning held in the term "skipped" is that _it was never deployed_ (but was rather _skipped_ over in favor of another version) — superseded does not carry that connotation and will likely be perceived as meaning the same thing as previous (which we call "decommissioned" in deck) when folks are looking at it on a UI.

Reply:
not to say that that has anything to do with whether it's useful as a naming scheme in keel's codebase

Reply:
I think "skipped" is better in the UI, and was fine even as a status enum value. I just suggested that the term in the `skippedBy` and `skippedAt` fields could be replaced with "superseded" because it sounded weird in that context.

Reply:
Ooo I see. I have no opinion on this - is it worth me changing the enum back to `SKIPPED` then, and just using `skippedBy` and `skippedAt` instead of superseded everywhere?

Reply:
I really have no stake / opinion on this code specifically, so I'll let ya'll decide how you feel 😄

Reply:
I'm a big fan of consistency in terminology between implementation in the code and what's exposed in the UI (I was very influenced by Eric Evans's book _Domain-Driven Design_). If the UI is going to use the term "skipped", I think that the code should as well.

Reply:
Okay, I'll do that then! Thanks for weighing in.